src/main/kotlin/lotto/LottoNumber.kt

```diff
@@ -0,0 +1,18 @@
+package lotto
+@JvmInline
+value class LottoNumber(val number: Int) {
+
+    companion object {
+        private const val MIN_NUMBER = 1
+        private const val MAX_NUMBER = 46
+        val LOTTO_NUMBERS = (MIN_NUMBER until MAX_NUMBER).map { LottoNumber(it) }
+
+        fun valueOf(input: Int): LottoNumber {
+            if (input in MIN_NUMBER until MAX_NUMBER) {
+                return LottoNumber(input)
+            } else {
+                throw IllegalArgumentException("Invalid LottoNumber: $input. $MIN_NUMBER and ${MAX_NUMBER - 1} 사이의 값이여야 합니다.")
+            }
+        }
```

Comment:
미리 LottoNumber를 만들어두셨네요 👍 그런데 외부에서 LottoNumber(999) 이런식으로 만들어도 막을수가 없어보이는데 이 부분도 고려되면 좋을 것 같아요!

그리고 캐싱까지 고려해보면 좋을 것 같네요. 
```suggestion
        fun valueOf(input: Int): LottoNumber {
            return cacheNumbers[input] ?: throw IllegalArgumentException()
        }
```